### PR TITLE
fix(font): fix get glyph id bug.

### PR DESCRIPTION
**Pull request by Allen-azh**
For more details, see [Font Converter Error #7372](https://github.com/lvgl/lvgl/issues/7372)

### CODE REVIEW

File: src/font/lv_font_fmt_txt.c [hunk 1]
```diff
@@ -270,6 +270,7 @@ static uint32_t get_glyph_dsc_id(const lv_font_t * font, uint32_t letter)
         }
         else if(fdsc->cmaps[i].type == LV_FONT_FMT_TXT_CMAP_FORMAT0_FULL) {
             const uint8_t * gid_ofs_8 = fdsc->cmaps[i].glyph_id_ofs_list;
+            if(0 == gid_ofs_8[rcp] && letter != fdsc->cmaps[i].range_start) continue;
```
Comment (FASTSHIFT) [inline src/font/lv_font_fmt_txt.c]: Please use `gid_ofs_8[rcp] == 0` make it readable. Modern compilers can also warn about misspelled equals signs.

**Comment by FASTSHIFT** [comment]
cc @W-Mai

File: src/font/lv_font_fmt_txt.c (hunk 1 above)
Comment (kisvegabor) [inline src/font/lv_font_fmt_txt.c]:
```suggestion
            if(gid_ofs_8[rcp] == 0 && letter != fdsc->cmaps[i].range_start) continue;
```

Comment (kisvegabor) [reply src/font/lv_font_fmt_txt.c]: I've just fixed it.

**Comment by kisvegabor** [comment]
The proposed solution seems more like a workaround but a not a real solution to the problem. Does it work with a font converted to C? 

I've taken a deeper look and it seems `glyph_id_ofs_list` is generated incorrectly. These `0`s in the array look bad. It should be like 0, 1, 2, 3, 4, 5 ...
image

**Comment by Allen-azh** [comment]
> it work with a font converted to C

Yes, it work with a font converted to C.

**Comment by Allen-azh** [comment]
> The proposed solution seems more like a workaround but a not a real solution to the problem. Does it work with a font converted to C?提出的解决方案似乎更像是一种解决方法，但并不是问题的真正解决方案。它适用于转换为 C 的字体吗？
> 
> I've taken a deeper look and it seems `glyph_id_ofs_list` is generated incorrectly. These `0`s in the array look bad. It should be like 0, 1, 2, 3, 4, 5 ...我仔细研究了一下，它似乎 `glyph_id_ofs_list` 生成不正确。数组中的 `0` s 看起来很糟糕。它应该是 0、1、2、3、4、5 ... image

Yes, it work with a font converted to C.

File: src/font/lv_font_fmt_txt.c [hunk 2]
```diff
@@ -270,6 +270,7 @@ static uint32_t get_glyph_dsc_id(const lv_font_t * font, uint32_t letter)
         }
         else if(fdsc->cmaps[i].type == LV_FONT_FMT_TXT_CMAP_FORMAT0_FULL) {
             const uint8_t * gid_ofs_8 = fdsc->cmaps[i].glyph_id_ofs_list;
+            if(gid_ofs_8[rcp] == 0 && letter != fdsc->cmaps[i].range_start) continue;
```
Comment (kisvegabor) [inline src/font/lv_font_fmt_txt.c:276]:
```suggestion
            /* The first character is always valid and should have offset = 0
             * However if a character is missing it also has offset=0.
             * So if there is a 0 not on the first position then it's a missing character */
            if(gid_ofs_8[rcp] == 0 && letter != fdsc->cmaps[i].range_start) continue;
```

**Comment by kisvegabor** [comment]
I've started to Debug it and finally understood the issue. The first glyph is mapped to index=0, but the missing characters are also mapped to 0. In light of that, I think your suggestion is correct because:
1. The first character is always valid and should have index=0
2. index=0 later always means a missing character

I've added this as a comment in the code.

**Review by kisvegabor — APPROVED** [review]

**Comment by Allen-azh** [comment]
Hi,

Thank you for your feedback and for understanding the issue! I'm glad my suggestion helped resolve the problem. I also appreciate you adding the comment in the code; it makes things much clearer. I hope this change effectively resolves the issue. Feel free to reach out if there are more opportunities for improvements.

Thanks again for your effort and support!

**Comment by Allen-azh** [comment]
@FASTSHIFT The issue has been resolved. Could you kindly review it at your earliest convenience?

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thank you @Allen-azh% :blush:

**Pull request merged**